provider/cloudflare/cloudflare.go

```diff
@@ -190,6 +195,45 @@ func (z zoneService) CreateCustomHostname(ctx context.Context, zoneID string, ch
 	return z.service.CreateCustomHostname(ctx, zoneID, ch)
 }
 
+type DNSRecordsConfig struct {
+	PerPage int
+	Comment string
+}
+
+func (c *DNSRecordsConfig) trimAndValidateComment(dnsName, comment string, paidZone func(string) bool) string {
+	if len(comment) > freeZoneMaxCommentLength {
+		if !paidZone(dnsName) {
+			log.Warnf("DNS record comment is invalid. Trimming comment of %s. To avoid endless syncs, please set it to less than %d chars.", dnsName, freeZoneMaxCommentLength)
+			return comment[:freeZoneMaxCommentLength]
+		} else if len(comment) > paidZoneMaxCommentLength {
+			log.Warnf("DNS record comment is invalid. Trimming comment of %s. To avoid endless syncs, please set it to less than %d chars.", dnsName, paidZoneMaxCommentLength)
+			return comment[:paidZoneMaxCommentLength]
+		}
+	}
```

Comment:
not required to make any changes. depends on readability, but sometimes it's easier to read when there is a reduced nesting

example

```go
if len(comment) <= freeZoneMaxCommentLength { return # early return }
if / else {}
```

which will be something like

```go
if len(comment) <= freeZoneMaxCommentLength {
	return comment
}

maxLength := freeZoneMaxCommentLength
if paidZone(dnsName) {
	maxLength = paidZoneMaxCommentLength
}

if len(comment) > maxLength {
	log.Warnf("DNS record comment is invalid. Trimming comment of %s. To avoid endless syncs, please set it to less than %d chars.", dnsName, maxLength)
	return comment[:maxLength]
}

return comment
```